OpenEdX/Managers/DeepLinkManager/DeepLinkManager.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  DeepLinkManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 24/01/2024.
+//
+
+import Foundation
+import Core
+import UIKit
+
+public protocol DeepLinkService {
+    func configureWith(launchOptions: [UIApplication.LaunchOptionsKey: Any]?)
+    func handledURLWith(app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any]) -> Bool
+}
+
+class DeepLinkManager {
+    private var services: [DeepLinkService] = []
+    
+    // Init manager
+    public init(config: ConfigProtocol) {
+        services = servicesFor(config: config)
+    }
+    
+    private func servicesFor(config: ConfigProtocol) -> [DeepLinkService] {
+        var deepServices: [DeepLinkService] = []
```

Comment:
We are not going with method name renaming then how about renaming it to services?

Reply:
It's just a `var` to return from a function. Why is its name so important if you are asking me to rename it a second time?

Reply:
I believe the names should be meaningful according to their usage. It doesn't matter whether they are local or instance variables.

Reply:
It can be renamed to services or deeplinkServices as per the usage.

Reply:
Do you see a big difference between `deepServices` and `deeplinkServices`? Is it not "meaningful" enough?
Here https://github.com/openedx/openedx-app-ios/pull/263#discussion_r1481674064 you asked me to rename to 'pushProviders' - why not to 'pushnotificationProviders' to ' be meaningful according to their usage'? I'm just following your example.
 Don't you think we're just wasting time here?

Reply:
I believe var `deepServices` of type `[DeepLinkService]` is enough 'readable' (especially since it's not used anywhere but these 5 lines of code)

Reply:
hmm ok if you are reluctant to make this change, then lets go with it.